core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectLifecycle.scala

```diff
@@ -99,4 +102,36 @@ class BpmBillingProjectCreator(billingRepository: BillingRepository,
     } catch {
       case exception: Exception => Future.failed(exception)
     }
+
+  override def preDeletionSteps(projectName: RawlsBillingProjectName, ctx: RawlsRequestContext): Future[Unit] =
+    for {
+      _ <- billingRepository.getCreationStatus(projectName).flatMap {
+        case CreationStatuses.CreatingLandingZone =>
+          Future.failed(
+            new BillingProjectDeletionException(
+              RawlsErrorReport(
+                s"Billing project ${projectName.value} cannot be deleted because its landing zone is still being created"
+              )
+            )
+          )
+        case _ => Future.successful()
+      }
+      _ <- billingRepository.getLandingZoneId(projectName).flatMap {
```

Comment:
Nit: If you change this from `.flatMap` to `.map`, you won't need to return `Future.successful()`

You're already throwing the exception directly when you get the errorReport, and you're blocking on the call to WSM to delete the landing zone, so you aren't really getting any use out of `.flatMap`

Reply:
@blakery so do I need `flatMap` at line 108, or can that one also be simplified? I'm again confused about map vs. flatMap and mostly was just excited to figure out a syntax that works.

Reply:
If I replace `flatMap` with `map` I won't be able to log the earning about the landing zone not existing. Perhaps not a big deal, but I thought it was good to note because it isn't "expected".

Reply:
The explanation above applies here too. 
You will never get `None` here, because `getLandingZoneId` throws an exception there - so you'd want `recover` or `recoverWith` to log it.

Reply:
@blakery I think this one is different-- None is possible in this case, meaning that the billing project existed, but the landingZoneId was None. The exception that is thrown by `billingRepository.getLandingZoneId` is when the billing project doesn't exist.

The type of `landingZoneId` is Option[String], whereas the type of `status` is CreationStatuses.CreationStatus.

Reply:
We talked, and I misunderstood what Blake was suggested. New code has been pushed up.